src/ToricVarieties/ToricLineBundles/attributes.jl

```diff
@@ -60,9 +113,33 @@ export degree
 @doc Markdown.doc"""
     basis_of_global_sections_via_rational_functions(l::ToricLineBundle)
 
-Return a basis of the global sections of the toric line bundle `l`.
+Return a basis of the global sections of the toric line bundle `l` in terms of rational functions.
+
+# Examples
+```jldoctest
+julia> v = projective_space(NormalToricVariety, 2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> l = ToricLineBundle(v, [fmpz(2)])
+A toric line bundle on a normal toric variety
+
+julia> basis_of_global_sections_via_rational_functions(l)
+6-element Vector{MPolyQuoElem{fmpq_mpoly}}:
+ x1_^2
+ x2*x1_^2
+ x2^2*x1_^2
+ x1_
+ x2*x1_
+ 1
+```
 """
 @attr function basis_of_global_sections_via_rational_functions(l::ToricLineBundle)
```

Comment:
Is this always meant to return a `Vector{MPolyQuoElem{fmpq_mpoly}}`? Then perhaps indicate this:
```suggestion
@attr Vector{MPolyQuoElem{fmpq_mpoly}} function basis_of_global_sections_via_rational_functions(l::ToricLineBundle)
```
Or at least indicate that a vector is returned:
```suggestion
@attr Vector function basis_of_global_sections_via_rational_functions(l::ToricLineBundle)
```

Reply:
```julia
julia> basis_of_global_sections(ToricLineBundle(del_pezzo(3),[4,0,0,0]))
15-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
 e2^4*x3^4*e3^4
 e1*x2*e2^3*x3^3*e3^4
 e1^2*x2^2*e2^2*x3^2*e3^4
 e1^3*x2^3*e2*x3*e3^4
 e1^4*x2^4*e3^4
 x1*e1*e2^4*x3^3*e3^3
 x1*e1^2*x2*e2^3*x3^2*e3^3
 x1*e1^3*x2^2*e2^2*x3*e3^3
 x1*e1^4*x2^3*e2*e3^3
 x1^2*e1^2*e2^4*x3^2*e3^2
 x1^2*e1^3*x2*e2^3*x3*e3^2
 x1^2*e1^4*x2^2*e2^2*e3^2
 x1^3*e1^3*e2^4*x3*e3
 x1^3*e1^4*x2*e2^3*e3
 x1^4*e1^4*e2^4
```
It could also be `fmpz` instead of `fmpq`, if the user opts to use `ZZ` as coefficient ring rather than `QQ`.

I guess I will add vector as return type, as you suggest. This applies equally to `basis_of_global_sections` and `basis_of_global_sections_via_homogeneous_component`.

Reply:
I suggest to fix this in a separate commit and have thus added this to the issue in https://github.com/oscar-system/Oscar.jl/issues/1257.